core/src/main/scala/tapir/Codec.scala

```diff
@@ -267,7 +293,7 @@ object CodecForMany {
         case List(h) => tm.decode(h).map(Some(_))
         case l       => DecodeResult.Multiple(l)
       }
-      override val meta: CodecMeta[M, R] = tm.meta.copy(isOptional = true)
+      override val meta: CodecMeta[Option[T], M, R] = tm.meta.copy(isOptional = true, validator = tm.meta.validator.toOption)
```

Comment:
I would expect `.toOption` to convert this to an `Option[Valdiator]` or sth like that :) Maybe `.forOption` and `.forIterable` instead?